### PR TITLE
Alloca serialisation tweaks.

### PR DESCRIPTION
**Pull request by vext01**
Reject allocas we don't support and include alignment in the lowered instruction.

### CODE REVIEW

File: llvm/lib/YkIR/YkIRWriter.cpp [hunk 1]
```diff
@@ -450,6 +450,18 @@ class YkIRWriter {
 
   void serialiseAllocaInst(AllocaInst *I, FuncLowerCtxt &FLCtxt, unsigned BBIdx,
                            unsigned &InstIdx) {
+    // We don't yet support:
+    //  - the `inalloca` keyword.
+    //  - non-zero address spaces.
+    //  - dynamic alloca (because stackmaps can't handle them).
+    //  - allocating a number of objects that doesn't fit in a `size_t`.
```
Comment (ptersilie) [inline llvm/lib/YkIR/YkIRWriter.cpp:457]: s/doesn't/don't

**Comment by ptersilie** [comment]
Just one comment.

**Comment by vext01** [comment]
Try that.

**Comment by ptersilie** [comment]
Love it! Please squash,

**Comment by vext01** [comment]
squashed.

**Pull request merged**